lib/request-helpers.js

```diff
@@ -84,3 +84,30 @@ const setCacheControlForStatsResponse = (res, filter) => {
     res.setHeader('cache-control', `public, max-age=${365 * 24 * 3600}, immutable`)
   }
 }
+
+// Fetch the "day" (DATE) as a string (TEXT) to prevent node-postgres from converting it into
+// a JavaScript Date with a timezone, as that could change the date one day forward or back.
+export const getDailyDistinctCountQuery = (table, column, asColumn = null) => {
+  if (!asColumn) asColumn = column + '_count'
+  return `
+    SELECT day::TEXT, COUNT(DISTINCT ${column})::INT as ${asColumn}
+    FROM ${table}
```

Comment:
this brings an SQL injection risk, if we're not careful about passing user input into this function. What do you think about instead of return SQL strings here, we also perform the query? The `pg` library will protect us from injections.

```
// platform-stats-fetchers.js
const rows = await getMonthlyDistinctCount({ pgPool, table: 'daily_stations', column: 'statiton_id', from: filter.from, to: filter.to })
```

Reply:
I don't expect the table name to be ever provided by the user, but I also agree it's better to be careful.

Let's use [`pg.escapeIdentifier`](https://node-postgres.com/apis/utilities#pgescapeidentifier) to create values that are safe to be included in the SQL query string.

I find it problematic that this function returns the parameterized query but not the parameters. IMO, the function should either return both (query string + parameters array) or execute the query.

Reply:
Agree with all, addressed in https://github.com/filecoin-station/spark-stats/pull/93/commits/cf9be9cec511e2610598f8a832dde38fbb5f6008